src/Disks/ReadIndirectBufferFromRemoteFS.cpp

```diff
@@ -105,12 +108,21 @@ bool ReadIndirectBufferFromRemoteFS<T>::nextImpl()
     const auto & path = metadata.remote_fs_objects[current_buf_idx].first;
 
     current_buf = createReadBuffer(path);
-    current_buf->next();
 
-    working_buffer = current_buf->buffer();
-    absolute_position += working_buffer.size();
+    return nextAndShiftPosition();
+}
+
+template <typename T>
+bool ReadIndirectBufferFromRemoteFS<T>::nextAndShiftPosition()
+{
+    swap(*current_buf);
+    auto result = current_buf->next();
+    swap(*current_buf);
+
+    if (result)
+        absolute_position += working_buffer.size();
 
-    return true;
+    return result;
```

Comment:
Could you please add comment about why we should act like this, maybe something about what each buffer does.